templates/CRM/Contribute/Form/Contribution.tpl

```diff
@@ -488,7 +488,9 @@
      });
 
      function showHideCancelInfo(obj) {
-       if (obj.find(":selected").text() == 'Refunded' || obj.find(":selected").text() == 'Cancelled') {
+       var selected_status_id  = obj.find(":selected").val();
```

Comment:
It could, but it seemed more readable to me this way. But if you prefer a different style, paste your code and I'll put it in...